horace_core/sqw/coord_transform/@CurveProjBase/private/check_and_set_type_.m

```diff
@@ -13,12 +13,17 @@
 
 for i= 1:3
     let = lower(val(i));
-    if ~ismember(let,obj.types_available_{i})
-        error('HORACE:sphere_proj:invalid_argument',...
+    if ~ismember(let,obj.curve_proj_types_{i})
+        error('HORACE:CurveProjBase:invalid_argument',...
             'letter "%s" does not belong to the available types: ("%s") for axis N_%d',...
-            let,disp2str(obj.types_available_{i}),i);
+            let,disp2str(obj.curve_proj_types_{i}),i);
     end
     val(i) = let;
 end
-
-obj.type_ = val;
+if ~isequal(obj.type_,val)
+    obj.type_             = val;
+    obj.img_scales_cache_ = [];
```

Comment:
checking that img_scales_cache is defined for the else case

Reply:
like in any `serializable` caches are calculated in `check_combo_arg` as they may depend on mulitple properties. 
In setter we just invalidate existing cahce if it depends on this property. 

The same design as suggested for `hashable`